tests/test_regression.py

```diff
@@ -0,0 +1,124 @@
+"""
+from pathlib import Path
+from subprocess import run
+from urllib.request import urlretrieve
+
+import numpy as np
+import pytest
+from pineappl.fk_table import FkTable
+from yaml import dump, safe_load
+
+THEORIES_REPO = "https://raw.githubusercontent.com/NNPDF/theories/main"
+THEORYID = 400
+
+
+def _download_resources(filename, tmp_path):
+    """Download resources (filename) from the theories repo and put it in the same path
+    relative to THEORIES_REPO
+    """
+    output_file = tmp_path / filename
+    output_file.parent.mkdir(exist_ok=True)
+    urlretrieve(f"{THEORIES_REPO}/{filename}", output_file)
+    return output_file
+
+
+def _download_dataset(dataset, theoryid, tmp_path):
+    """Download both the yaml file and all grids for a given dataset for a given theory"""
+    yaml_file = f"data/yamldb/{theoryid}/{dataset}.yaml"
+    _download_resources(yaml_file, tmp_path)
+    # However, the yaml file goes into the "root" of the yaml directory, move it there!
+    right_yaml = tmp_path / "data" / "yamldb" / f"{dataset}.yaml"
+    (tmp_path / yaml_file).rename(right_yaml)
+    # Download the relevant grids for this dataset
+    res = safe_load(right_yaml.open("r", encoding="utf-8"))
+    grids = [i for j in res["operands"] for i in j]
+    for grid_name in grids:
+        _download_resources(f"data/grids/400/{grid_name}.pineappl.lz4", tmp_path)
+    return grids
+
+
+class _FakePDF:
+    """A Fake lhapdf-like PDF to evolve the grids"""
+
+    def __init__(self):
+        pids = np.arange(-6, 7, dtype=int)
+        pids[6] = 21
+
+        alphas = np.linspace(1.2, 1.8, len(pids))
+        betas = np.linspace(1.2, 3.8, len(pids))
+
+        self._alphas = dict(zip(pids, alphas))
+        self._betas = dict(zip(pids, betas))
+
+    def xfxQ2(self, pid, x, q2):
+        """Compute x^alpha*(1-x)^beta"""
+        alpha = self._alphas[pid]
+        beta = self._betas[pid]
+        return np.power(x, alpha) * np.power(1 - x, beta)
+
+
+def _trim_template(template_card, take_points=10):
+    """Trim the template card so that the number of x-values to compute is much smaller"""
+    card_info = safe_load(template_card.open("r", encoding="utf-8"))
+    original_x = np.array(card_info["rotations"]["xgrid"])
+    skip = int(len(original_x) / take_points)
+    idxs = list(range(0, len(original_x), skip))
+    card_info["rotations"]["xgrid"] = original_x[idxs].tolist()
+    dump(card_info, template_card.open("w", encoding="utf-8"))
+
+
+@pytest.mark.parametrize("dataset", ["LHCBWZMU8TEV", "INTEGXT3"])
+def test_regression(tmp_path, dataset, rebuild=False):
+    """Run pineko through subprocess to ensure that the shell scripts are working exactly
+    as intended.
+    """
+    # We start by downloading pineko.toml in order to generate the folder structure
+    _download_resources("pineko.toml", tmp_path)
+    # Which we create... now!
+    run(["pineko", "scaffold", "new"], cwd=tmp_path, check=True)
```

Comment:
Pineko functions can be run programmatically, without the need of `subprocess`es.
The CLI has been designed to be a thin wrapper over functions (as all the others in the pineline, starting with Pinefarm), such that you are actually able to script them, within Python.

If this is not possible, I'd ask you the favor of filing an issue!